src/wallet/direct_send.rs

```diff
@@ -72,11 +72,28 @@ impl Wallet {
         coins_to_spend: &[(ListUnspentResultEntry, UTXOSpendInfo)],
     ) -> Result<Transaction, WalletError> {
         log::info!("Creating Direct-Spend from Wallet.");
-        let mut tx_inputs = Vec::<TxIn>::new();
-        let mut spend_infos = Vec::new();
+
+        // Set the Anti-Fee-Snipping locktime
+        let current_height = self.rpc.get_block_count()?;
+        let lock_time = LockTime::from_height(current_height as u32)?;
+
+        let mut tx = Transaction {
+            version: Version::TWO,
+            lock_time,
+            input: vec![],
+            output: vec![],
+        };
+
         let mut total_input_value = Amount::ZERO;
 
         for (utxo_data, spend_info) in coins_to_spend {
+            // ensure that no fedility utxos must come.
+            if let UTXOSpendInfo::FidelityBondCoin { .. } = spend_info {
+                return Err(WalletError::Protocol(
+                    "Fidelity Bonds cannot be spent directly.Use redeem_fidelity instead"
+                        .to_string(),
+                ));
+            }
```

Comment:
If we are checking why not check for ContactUtxos too? And instead of hard error why not just filter then out with an warning message and carry on creating transaction.